node/src/chain/fsm/outofsync.rs

```diff
@@ -227,13 +225,13 @@ impl<DB: database::DB, VM: vm::VMExecution, N: Network>
         blk: &Block,
         metadata: Option<Metadata>,
     ) -> anyhow::Result<bool> {
-        let mut acc = self.acc.write().await;
+        let acceptor = self.acc.clone();
+        let mut acc = acceptor.write().await;
         let block_height = blk.header().height;
 
-        if self.attempts == 0 && self.is_timeout_expired() {
-            acc.restart_consensus().await;
-            // Timeout-ed sync-up
-            // Transit back to InSync mode
+        // We check the timeout here to prevent the peer from keeping us in
+        // outofsync by flooding our node with blocks
+        if self.on_sync_timeout().await {
```

Comment:
On_sync_timeout try to acquire a write lock from the acceptor (but you already acquired it, so probably it get stuck)

Try this before merge

Reply:
Will do!